packages/wdio-reporter/src/index.ts

```diff
@@ -207,19 +207,37 @@ export default class WDIOReporter extends EventEmitter {
             }
         })
 
+        /**
+         * deduplicate long script strings (> 1KB) to avoid memory leaks.
+         */
+        const scriptMap = new Map<string, string>()
+        const deduplicateScript =
+            (payload: { body: { script?: string } }) => {
+                const script = payload.body?.script
+                if (typeof script === 'string' && script.length > 1000) { // bytes
```

Comment:
`1000` seems like a magic number, maybe we can make it a const and add a comment why we decide to pick this